evaluations/centrifuge-gsrpc-v2_2_niklabh.md

```diff
@@ -0,0 +1,574 @@
+| 3.     | PR with changes                  | <ul><li>[ ] </li> </ul> | [centrifuge/go-substrate-rpc-client#338](https://github.com/centrifuge/go-substrate-rpc-client/pull/338)                                                                          | PR looks good                                                                                                             |
+| 4a.    | Registry Unit Tests              | <ul><li>[ ] </li> </ul> | [UnitTest](https://github.com/centrifuge/go-substrate-rpc-client/blob/20047b8b15d046d0fa137cc716cb489395a0081a/registry/registry_test.go)                                         | Unit Test coverage is good                                                                                                |
+| 4b.    | Registry (Events) Live Tests     | <ul><li>[ ] </li> </ul> | [Events Live Tests](https://github.com/centrifuge/go-substrate-rpc-client/blob/20047b8b15d046d0fa137cc716cb489395a0081a/registry/retriever/event_retriever_live_test.go)          | Error in running docker tests, submitted a [PR](https://github.com/centrifuge/go-substrate-rpc-client/pull/353) to fix    |
+| 4c.    | Registry (Extrinsics) Live Tests | <ul><li>[ ] </li> </ul> | [Extrinsics Live Tests](https://github.com/centrifuge/go-substrate-rpc-client/blob/20047b8b15d046d0fa137cc716cb489395a0081a/registry/retriever/extrinsic_retriever_live_test.go)  | Live test against public endpoints. Tests are running for too long. No need to check for 1000 events. 50 are fine         |
+| 4d.    | Test execution                   | <ul><li>[x] </li> </ul> | [Test execution](https://github.com/centrifuge/go-substrate-rpc-client/blob/20047b8b15d046d0fa137cc716cb489395a0081a/Makefile#L49-L53)                                            | Makefile has a [error](https://github.com/centrifuge/go-substrate-rpc-client/issues/352)                                  |
+| 5.     | Dockerfile                       | <ul><li>[x] </li> </ul> | [Dockerfile](https://github.com/centrifuge/go-substrate-rpc-client/blob/b9da36ebf177cbb3ee41c2ef2b9a1fb08e5f8522/Dockerfile_milestone2)                                           | Dockerfile is provided                                                                                                    |
+| 6.     | In-Memory Registry               | <ul><li>[x] </li> </ul> | [centrifuge/go-substrate-rpc-client#338](https://github.com/centrifuge/go-substrate-rpc-client/pull/338)                                                                          | In-Memory Registry is implemented                                                                                         |
+
+## General Notes v1 - June 26, 2023
+
+- Documentation needs to be improved. Basic tutorial on how to parse any substrate event and type using the new version of the library is missing.
```

Comment:
The `retriever.GetEvents(blockHash)` is the one in charge of parsing the events of a block into the `parser.Event` struct. How to use it is described [here](https://github.com/centrifuge/go-substrate-rpc-client/blob/20047b8b15d046d0fa137cc716cb489395a0081a/registry/retriever/event_retriever_live_test.go#L70) as stated in the REGISTRY.md. 
Are you referring to the deserialization/unmarshalling of those events into a user defined (strongly typed) structs?

Reply:
OK! Thanks @semuelle. Added the changes requested in this [commit](https://github.com/centrifuge/go-substrate-rpc-client/commit/7e3e5ad5225bfa7afd342f3fe8932dc44702c472) and updated the delivery.